torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -70,6 +71,29 @@ std::vector<c10::DeviceIndex> getDevicesForTensors(
   }
 }
 
+std::vector<at::cuda::CUDAStream> getStreamsForTpMessage(
+    const tensorpipe::Message& tpMessage) {
+  std::vector<at::cuda::CUDAStream> streams;
+  for (auto& tensor : tpMessage.tensors) {
```

Comment:
We're getting one stream per tensor, even if multiple tensor could be on the same device. Could it make sense instead to just get one stream per device, regardless of how many tensors we have and where they are?

Reply:
Yep, the plan is to get one stream per device. That part is not ready for review yet.